apps/api/src/auth/auth.controller.ts

```diff
@@ -220,4 +228,49 @@ export class AuthController {
 
     response.redirect(this.APP_URL);
   }
+
+  @Post("mfa/setup")
+  @Roles(...Object.values(USER_ROLES))
+  @Validate({
+    response: baseResponse(mfaSetupResponseSchema),
+  })
+  async mfaSetup(@CurrentUser("userId") userId: UUIDType) {
+    const user = await this.userService.getUserById(userId);
+
+    if (!user) {
+      throw new UnauthorizedException("User not found");
+    }
+
+    const secret = await this.authService.generateMFASecret(user.id);
+
+    return new BaseResponse({
+      secret,
+      otpauth: `otpauth://totp/Mentingo:${user.email}?secret=${secret}&issuer=Mentingo`,
+    });
+  }
+
+  @Post("mfa/verify")
+  @Roles(...Object.values(USER_ROLES))
+  @Validate({
+    request: [{ type: "body", schema: mfaVerifySchema }],
+    response: baseResponse(mfaVerifyResponseSchema),
+  })
+  async mfaVerify(
+    @Body() body: MFAVerifyBody,
+    @CurrentUser("userId") userId: UUIDType,
+    @Res({ passthrough: true }) response: Response,
+  ) {
+    const mfaVerifyResult = await this.authService.verifyMFACode(userId, body.token);
+
+    if (!mfaVerifyResult) {
+      return new BaseResponse({ isValid: false });
+    }
+
+    const { isValid, accessToken, refreshToken } = mfaVerifyResult;
+
+    this.tokenService.clearTokenCookies(response);
+    this.tokenService.setTokenCookies(response, accessToken, refreshToken, true);
```

Comment:
This should be moved out from the controller method. Most likely to new method in `authService`